src/main/java/appeng/items/misc/ItemEncodedPattern.java

```diff
@@ -109,6 +110,42 @@ public void addCheckedInformation( final ItemStack stack, final World world, fin
 		if( details == null )
 		{
 			lines.add( TextFormatting.RED + GuiText.InvalidPattern.getLocal() );
+
+			try
+			{
+				InvalidPatternHelper invalid = new InvalidPatternHelper( stack, world );
+
+				final String label = ( invalid.isCraftable() ? GuiText.Crafts.getLocal() : GuiText.Creates.getLocal() ) + ": ";
+				final String and = ' ' + GuiText.And.getLocal() + ' ';
+				final String with = GuiText.With.getLocal() + ": ";
+
+				boolean first = true;
+				for( final String output : invalid.getOutputs() )
+				{
+					lines.add( ( first ? label : and ) + output );
+					first = false;
+				}
+
+				first = true;
+				for( final String input : invalid.getInputs() )
+				{
+					lines.add( ( first ? with : and ) + input);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)